### PR TITLE
Convert hab_install to a custom resource

### LINKED ISSUE #13: convert hab_install to a custom resource
Per @thommay in #11, we should use a custom resource for `hab_install`. The proposed `hab_service` in that PR, and the `hab_package` resource will stay as "library" resources because they subclass `Chef::Resource::Service` and `Chef::Resource::Package`, respectively.

### PR DESCRIPTION
**Pull request by jtimberman**
Closes #13

Signed-off-by: Joshua Timberman <joshua@chef.io>

### CODE REVIEW

File: resources/install.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#
+include HabitatCookbook::Helpers
+
+resource_name :hab_install
+provides :hab_install
+
+property :install_url, String, default: "https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh"
+
+default_action :install
+allowed_actions :install, :upgrade
```
Comment (tas50) [inline resources/install.rb]: we don't need either of these

File: resources/install.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+action :install do
+  return if ::File.exist?(hab_path)
+
+  remote_file ::File.join(Chef::Config[:file_cache_path], "/hab-install.sh") do
```
Comment (tas50) [inline resources/install.rb]: if we're doing a file join do we need this to be /hab-install.sh vs. hab-install.sh?

File: resources/install.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+resource_name :hab_install
+provides :hab_install
+
+property :install_url, String, default: "https://raw.githubusercontent.com/habitat-sh/habitat/master/components/hab/install.sh"
+
+default_action :install
+allowed_actions :install, :upgrade
+
+action :install do
+  return if ::File.exist?(hab_path)
```
Comment (tas50) [inline resources/install.rb]: Where is hab_path coming from now that the method is gone?

File: resources/install.rb (hunk 2 above)
Comment (jtimberman) [inline resources/install.rb]: Deleted the /

File: resources/install.rb (hunk 3 above)
Comment (jtimberman) [inline resources/install.rb]: It's back! In the resource file now instead of a helper.

**Pull request merged**